packages/connect/src/device/Device.ts

```diff
@@ -595,6 +606,47 @@ export class Device extends TypedEmitter<DeviceEvents> {
         }
     }
 
+    async checkFirmwareHash(): Promise<FirmwareHashCheckResult> {
+        const createFailResult = (error: FirmwareHashCheckError) => ({ success: false, error });
+
+        // optional setting, see types/settings.ts
+        const baseUrl = DataManager.getSettings('binFilesBaseUrl');
+        if (!baseUrl) return createFailResult('binary-unavailable');
+
+        const checkSupported = this.atLeast(FW_HASH_SUPPORTED_VERSIONS);
```

Comment:
if `FW_HASH_SUPPORTED_VERSIONS` is used only here, I would move definition to this place.

Reply:
It's also used in `onCallFirmwareUpdate`, where I extracted it from, so I can't place it in this file, because it'd create circular dependency. Placing it in `onCallFirmwareUpdate` would work, but does not make much sense semantically.
:bulb:  But I noticed there is `src/constants/` folder, so I'll create `src/constants/device.ts`. Does that sound good?